The business intelligence microservice throws errors on docker.

It maybe related to PostgreSQL

<img width="1420" alt="screen shot 2017-01-30 at 22 44 49" src="https://cloud.githubusercontent.com/assets/10467382/22443253/912c1f38-e73e-11e6-96ce-f59eb81d0d11.png">

<img width="1419" alt="screen shot 2017-01-30 at 22 46 40" src="https://cloud.githubusercontent.com/assets/10467382/22443256/938ce8b6-e73e-11e6-8ee0-6ab408c397c4.png">

Hibernate: select count(databaseda0_.id) as col_0_0_ from database_data_source databaseda0_ inner join data_source databaseda0_1_ on databaseda0_.id=databaseda0_1_.id
Hibernate: select databaseda0_.id as id1_7_, databaseda0_1_.created_by_id as created_2_7_, databaseda0_1_.created_date as created_3_7_, databaseda0_1_.last_modified_by_id as last_mod4_7_, databaseda0_1_.last_modified_date as last_mod5_7_, databaseda0_1_.type as type6_7_, databaseda0_1_.description as descript7_7_, databaseda0_1_.filter as filter8_7_, databaseda0_1_.name as name9_7_, databaseda0_.database_connection_id as database3_15_, databaseda0_.selected_table as selected1_15_ from database_data_source databaseda0_ inner join data_source databaseda0_1_ on databaseda0_.id=databaseda0_1_.id order by databaseda0_.id asc limit ?
2017-01-30 21:44:37 451 [WARN] [c=SqlExceptionHelper] SQL Error: 0, SQLState: 22003
2017-01-30 21:44:37 453 [ERROR] [c=SqlExceptionHelper] Bad value for type long : {"group": {"operator": "AND","rules": []}}
Jan 30, 2017 9:44:37 PM org.apache.catalina.core.StandardWrapperValve invoke
SEVERE: Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is org.springframework.dao.DataIntegrityViolationException: could not execute query; SQL [select databaseda0_.id as id1_7_, databaseda0_1_.created_by_id as created_2_7_, databaseda0_1_.created_date as created_3_7_, databaseda0_1_.last_modified_by_id as last_mod4_7_, databaseda0_1_.last_modified_date as last_mod5_7_, databaseda0_1_.type as type6_7_, databaseda0_1_.description as descript7_7_, databaseda0_1_.filter as filter8_7_, databaseda0_1_.name as name9_7_, databaseda0_.database_connection_id as database3_15_, databaseda0_.selected_table as selected1_15_ from database_data_source databaseda0_ inner join data_source databaseda0_1_ on databaseda0_.id=databaseda0_1_.id order by databaseda0_.id asc]; nested exception is org.hibernate.exception.DataException: could not execute query] with root cause
org.postgresql.util.PSQLException: Bad value for type long : {"group": {"operator": "AND","rules": []}}
at org.postgresql.jdbc2.AbstractJdbc2ResultSet.toLong(AbstractJdbc2ResultSet.java:2992)
at org.postgresql.jdbc2.AbstractJdbc2ResultSet.getLong(AbstractJdbc2ResultSet.java:2164)
at org.postgresql.jdbc2.AbstractJdbc2ResultSet.getClob(AbstractJdbc2ResultSet.java:437)
at org.postgresql.jdbc2.AbstractJdbc2ResultSet.getClob(AbstractJdbc2ResultSet.java:425)
at org.apache.commons.dbcp.DelegatingResultSet.getClob(DelegatingResultSet.java:568)
at org.apache.commons.dbcp.DelegatingResultSet.getClob(DelegatingResultSet.java:568)
at org.hibernate.type.descriptor.sql.ClobTypeDescriptor$1.doExtract(ClobTypeDescriptor.java:44)
at org.hibernate.type.descriptor.sql.BasicExtractor.extract(BasicExtractor.java:47)
at org.hibernate.type.AbstractStandardBasicType.nullSafeGet(AbstractStandardBasicType.java:238)
at org.hibernate.type.AbstractStandardBasicType.nullSafeGet(AbstractStandardBasicType.java:234)
at org.hibernate.type.AbstractStandardBasicType.nullSafeGet(AbstractStandardBasicType.java:224)
at org.hibernate.type.AbstractStandardBasicType.hydrate(AbstractStandardBasicType.java:300)
at org.hibernate.persister.entity.AbstractEntityPersister.hydrate(AbstractEntityPersister.java:2738)
at org.hibernate.loader.Loader.loadFromResultSet(Loader.java:1729)
at org.hibernate.loader.Loader.instanceNotYetLoaded(Loader.java:1655)
at org.hibernate.loader.Loader.getRow(Loader.java:1544)
at org.hibernate.loader.Loader.getRowFromResultSet(Loader.java:727)
at org.hibernate.loader.Loader.processResultSet(Loader.java:972)
at org.hibernate.loader.Loader.doQuery(Loader.java:930)
at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:336)
at org.hibernate.loader.Loader.doList(Loader.java:2617)
at org.hibernate.loader.Loader.doList(Loader.java:2600)
at org.hibernate.loader.Loader.listIgnoreQueryCache(Loader.java:2429)
at org.hibernate.loader.Loader.list(Loader.java:2424)
at org.hibernate.loader.hql.QueryLoader.list(QueryLoader.java:501)
at org.hibernate.hql.internal.ast.QueryTranslatorImpl.list(QueryTranslatorImpl.java:371)
at org.hibernate.engine.query.spi.HQLQueryPlan.performList(HQLQueryPlan.java:216)
at org.hibernate.internal.SessionImpl.list(SessionImpl.java:1326)
at org.hibernate.internal.QueryImpl.list(QueryImpl.java:87)
at org.hibernate.jpa.internal.QueryImpl.list(QueryImpl.java:606)
at org.hibernate.jpa.internal.QueryImpl.getResultList(QueryImpl.java:483)
at org.hibernate.jpa.criteria.compile.CriteriaQueryTypeQueryAdapter.getResultList(CriteriaQueryTypeQueryAdapter.java:50)
at org.springframework.data.jpa.repository.support.SimpleJpaRepository.readPage(SimpleJpaRepository.java:586)
at org.springframework.data.jpa.repository.support.SimpleJpaRepository.readPage(SimpleJpaRepository.java:566)
at org.springframework.data.jpa.repository.support.SimpleJpaRepository.findAll(SimpleJpaRepository.java:406)
at org.springframework.data.jpa.repository.support.SimpleJpaRepository.findAll(SimpleJpaRepository.java:375)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.executeMethodOn(RepositoryFactorySupport.java:503)
at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.doInvoke(RepositoryFactorySupport.java:488)
at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.invoke(RepositoryFactorySupport.java:460)
at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
at org.springframework.data.projection.DefaultMethodInvokingMethodInterceptor.invoke(DefaultMethodInvokingMethodInterceptor.java:61)
at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:99)
at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:281)
at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96)
at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:136)
at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:133)
at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:213)
at com.sun.proxy.$Proxy201.findAll(Unknown Source)
at com.abixen.platform.service.businessintelligence.multivisualization.service.impl.DatabaseDataSourceServiceImpl.findAllDataSources(DatabaseDataSourceServiceImpl.java:67)
at com.abixen.platform.service.businessintelligence.multivisualization.controller.DatabaseDataSourceController.getDataSources(DatabaseDataSourceController.java:54)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:221)
at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:136)
at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:114)
at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827)
at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738)
at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963)
at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:897)
at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:622)
at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:230)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.springframework.boot.web.filter.ApplicationContextHeaderFilter.doFilterInternal(ApplicationContextHeaderFilter.java:55)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.springframework.boot.actuate.trace.WebRequestTraceFilter.doFilterInternal(WebRequestTraceFilter.java:105)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317)
at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127)
at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:115)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:169)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:121)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.csrf.CsrfFilter.doFilterInternal(CsrfFilter.java:100)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:66)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214)
at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177)
at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346)
at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.springframework.web.filter.HttpPutFormContentFilter.doFilterInternal(HttpPutFormContentFilter.java:89)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:77)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.springframework.session.web.http.SessionRepositoryFilter.doFilterInternal(SessionRepositoryFilter.java:164)
at org.springframework.session.web.http.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:80)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.springframework.cloud.sleuth.instrument.web.TraceFilter.doFilter(TraceFilter.java:147)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.springframework.boot.actuate.autoconfigure.MetricsFilter.doFilterInternal(MetricsFilter.java:107)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198)
at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:108)
at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472)
at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:349)
at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:784)
at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:802)
at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1410)
at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
at java.lang.Thread.run(Thread.java:745)
2017-01-30T21:44:37.528338401Z 
Jan 30, 2017 9:44:37 PM zipkin.reporter.AsyncReporter$BoundedAsyncReporter$1 onError
WARNING: Dropped 2 spans due to ConnectException(Connection refused)
java.net.ConnectException: Connection refused
at java.net.PlainSocketImpl.socketConnect(Native Method)
at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
at java.net.Socket.connect(Socket.java:589)
at sun.net.NetworkClient.doConnect(NetworkClient.java:175)
at sun.net.www.http.HttpClient.openServer(HttpClient.java:432)
at sun.net.www.http.HttpClient.openServer(HttpClient.java:527)
at sun.net.www.http.HttpClient.<init>(HttpClient.java:211)
at sun.net.www.http.HttpClient.New(HttpClient.java:308)
at sun.net.www.http.HttpClient.New(HttpClient.java:326)
at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1169)
at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1105)
at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:999)
at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:933)
at sun.net.www.protocol.http.HttpURLConnection.getOutputStream0(HttpURLConnection.java:1283)
at sun.net.www.protocol.http.HttpURLConnection.getOutputStream(HttpURLConnection.java:1258)
at zipkin.reporter.urlconnection.URLConnectionSender.send(URLConnectionSender.java:174)
at zipkin.reporter.urlconnection.URLConnectionSender.sendSpans(URLConnectionSender.java:135)
at zipkin.reporter.AsyncReporter$BoundedAsyncReporter.flush(AsyncReporter.java:226)
at zipkin.reporter.AsyncReporter$Builder.lambda$build$0(AsyncReporter.java:153)
at zipkin.reporter.AsyncReporter$Builder$$Lambda$1.run(Unknown Source)
at java.lang.Thread.run(Thread.java:745)
2017-01-30T21:44:37.869236612Z

--- a/abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualization/controller/DatabaseDataSourceController.java
+++ b/abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualization/controller/DatabaseDataSourceController.java
@@ -20,6 +20,7 @@ import com.abixen.platform.core.util.ValidationUtil;
 import com.abixen.platform.core.util.WebModelJsonSerialize;
 import com.abixen.platform.service.businessintelligence.multivisualization.form.DatabaseDataSourceForm;
 import com.abixen.platform.service.businessintelligence.multivisualization.model.impl.datasource.database.DatabaseDataSource;
+import com.abixen.platform.service.businessintelligence.multivisualization.model.web.DataValueWeb;
 import com.abixen.platform.service.businessintelligence.multivisualization.model.web.DatabaseDataSourceWeb;
 import com.abixen.platform.service.businessintelligence.multivisualization.service.DatabaseDataSourceService;
 import com.fasterxml.jackson.annotation.JsonView;
@@ -33,6 +34,7 @@ import org.springframework.web.bind.annotation.*;
 
 import javax.validation.Valid;
 import java.util.List;
+import java.util.Map;
 
 
 @Slf4j
@@ -94,5 +96,14 @@ public class DatabaseDataSourceController {
         return new FormValidationResultDto(databaseDataSourceFormResult);
     }
 
+    @JsonView(WebModelJsonSerialize.class)
+    @RequestMapping(value = "/preview", method = RequestMethod.POST)
+    public  List<Map<String, DataValueWeb>> getPreviewData(@RequestBody @Valid DatabaseDataSourceForm databaseDataSourceForm) {
+        log.debug("createDataSource() - databaseDataSourceForm: " + databaseDataSourceForm);
+
+        List<Map<String, DataValueWeb>> databaseDataSourcePreviewData = databaseDataSourceService.getPreviewData(databaseDataSourceForm);
+
+        return databaseDataSourcePreviewData;
+    }
 
 }

--- a/abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualization/service/DatabaseDataSourceService.java
+++ b/abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualization/service/DatabaseDataSourceService.java
@@ -17,6 +17,7 @@ package com.abixen.platform.service.businessintelligence.multivisualization.serv
 import com.abixen.platform.service.businessintelligence.multivisualization.form.DatabaseDataSourceForm;
 import com.abixen.platform.service.businessintelligence.multivisualization.model.impl.datasource.DataSourceColumn;
 import com.abixen.platform.service.businessintelligence.multivisualization.model.impl.datasource.database.DatabaseDataSource;
+import com.abixen.platform.service.businessintelligence.multivisualization.model.web.DataValueWeb;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 
@@ -46,4 +47,6 @@ public interface DatabaseDataSourceService {
     DatabaseDataSource updateDataSource(DatabaseDataSource databaseDataSource);
 
     DatabaseDataSource findDataSource(Long id);
+
+    List<Map<String, DataValueWeb>> getPreviewData(DatabaseDataSourceForm databaseDataSourceForm);
 }

--- a/abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualization/service/DatabaseService.java
+++ b/abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualization/service/DatabaseService.java
@@ -39,4 +39,6 @@ public interface DatabaseService {
 
     List<Map<String, DataValueWeb>> getChartDataPreview(Connection connection, DatabaseDataSource databaseDataSource, ChartConfigurationForm chartConfigurationForm, String seriesName);
 
+    List<Map<String, DataValueWeb>> getDataSourcePreview(Connection connection, DatabaseDataSource databaseDataSource);
+
 }

--- a/abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualization/service/impl/AbstractDatabaseService.java
+++ b/abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualization/service/impl/AbstractDatabaseService.java
@@ -32,7 +32,8 @@ import java.util.stream.IntStream;
 
 public abstract class AbstractDatabaseService {
 
-    private static final int LIMIT = 6;
+    private final int chartLimit = 6;
+    private final int datasourceLimit = 20;
 
     @Autowired
     private JsonFilterService jsonFilterService;
@@ -142,7 +143,22 @@ public abstract class AbstractDatabaseService {
         }
         //FixMe
         List<Map<String, DataValueWeb>> data = getData(connection, databaseDataSource, chartColumnsSet, chartConfigurationForm);
-        return data.subList(0, data.size() < LIMIT ? data.size() : LIMIT);
+        return data.subList(0, data.size() < chartLimit ? data.size() : chartLimit);
+    }
+
+    public List<Map<String, DataValueWeb>> getDataSourcePreview(Connection connection, DatabaseDataSource databaseDataSource) {
+        Set<String> dataSourceColumnsSet = new HashSet<>();
+
+        databaseDataSource.getColumns().forEach(column -> {
+            dataSourceColumnsSet.add(column.getName());
+        });
+
+        if (dataSourceColumnsSet.isEmpty()) {
+            return new ArrayList<>();
+        }
+        //FixMe
+        List<Map<String, DataValueWeb>> data = getData(connection, databaseDataSource, dataSourceColumnsSet, null);
+        return data.subList(0, data.size() < datasourceLimit ? data.size() : datasourceLimit);
     }
 
     private List<Map<String, DataValueWeb>> getData(Connection connection, DatabaseDataSource databaseDataSource, Set<String> chartColumnsSet, ChartConfigurationForm chartConfigurationForm) {
@@ -151,17 +167,18 @@ public abstract class AbstractDatabaseService {
         try {
             Statement statement = connection.createStatement();
             ResultSetMetaData resultSetMetaData = getDatabaseMetaData(connection, databaseDataSource.getTable());
-            rs = statement.executeQuery(buildQueryForChartData(databaseDataSource, chartColumnsSet, resultSetMetaData, chartConfigurationForm));
-
-            if (rs != null) {
-                while (rs.next()) {
-                    final ResultSet row = rs;
-                    Map<String, DataValueWeb> rowMap = new HashMap<>();
-                    chartColumnsSet.forEach(chartColumnsSetElement -> {
-                        rowMap.put(chartColumnsSetElement, getDataFromColumn(row, chartColumnsSetElement));
-                    });
-                    data.add(rowMap);
-                }
+            if (chartConfigurationForm != null) {
+                rs = statement.executeQuery(buildQueryForChartData(databaseDataSource, chartColumnsSet, resultSetMetaData, chartConfigurationForm));
+            } else {
+                rs = statement.executeQuery(buildQueryForDataSourceData(databaseDataSource, chartColumnsSet, resultSetMetaData).toString());
+            }
+            while (rs.next()) {
+                final ResultSet row = rs;
+                Map<String, DataValueWeb> rowMap = new HashMap<>();
+                chartColumnsSet.forEach(chartColumnsSetElement -> {
+                    rowMap.put(chartColumnsSetElement, getDataFromColumn(row, chartColumnsSetElement));
+                });
+                data.add(rowMap);
             }
         } catch (SQLException e) {
             e.printStackTrace();
@@ -171,15 +188,20 @@ public abstract class AbstractDatabaseService {
     }
 
     private String buildQueryForChartData(DatabaseDataSource databaseDataSource, Set<String> chartColumnsSet, ResultSetMetaData resultSetMetaData, ChartConfigurationForm chartConfigurationForm) throws SQLException {
+        StringBuilder stringBuilder = buildQueryForDataSourceData(databaseDataSource, chartColumnsSet, resultSetMetaData);
+        stringBuilder.append(" AND ");
+        stringBuilder.append(jsonFilterService.convertJsonToJpql(chartConfigurationForm.getFilter(), resultSetMetaData));
+        return stringBuilder.toString();
+    }
+
+    private StringBuilder buildQueryForDataSourceData(DatabaseDataSource databaseDataSource, Set<String> chartColumnsSet, ResultSetMetaData resultSetMetaData) throws SQLException {
         StringBuilder stringBuilder = new StringBuilder("SELECT ");
         stringBuilder.append(chartColumnsSet.toString().substring(1, chartColumnsSet.toString().length() - 1));
         stringBuilder.append(" FROM ");
         stringBuilder.append(databaseDataSource.getTable());
         stringBuilder.append(" WHERE ");
         stringBuilder.append(jsonFilterService.convertJsonToJpql(databaseDataSource.getFilter(), resultSetMetaData));
-        stringBuilder.append(" AND ");
-        stringBuilder.append(jsonFilterService.convertJsonToJpql(chartConfigurationForm.getFilter(), resultSetMetaData));
-        return stringBuilder.toString();
+        return stringBuilder;
     }
 
     private DataValueWeb getDataFromColumn(ResultSet row, String columnName) {

--- a/abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualization/service/impl/DatabaseDataSourceServiceImpl.java
+++ b/abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualization/service/impl/DatabaseDataSourceServiceImpl.java
@@ -15,12 +15,12 @@
 package com.abixen.platform.service.businessintelligence.multivisualization.service.impl;
 
 import com.abixen.platform.service.businessintelligence.multivisualization.form.DatabaseDataSourceForm;
+import com.abixen.platform.service.businessintelligence.multivisualization.model.impl.database.DatabaseConnection;
 import com.abixen.platform.service.businessintelligence.multivisualization.model.impl.datasource.DataSourceColumn;
 import com.abixen.platform.service.businessintelligence.multivisualization.model.impl.datasource.database.DatabaseDataSource;
+import com.abixen.platform.service.businessintelligence.multivisualization.model.web.DataValueWeb;
 import com.abixen.platform.service.businessintelligence.multivisualization.repository.DatabaseDataSourceRepository;
-import com.abixen.platform.service.businessintelligence.multivisualization.service.DatabaseConnectionService;
-import com.abixen.platform.service.businessintelligence.multivisualization.service.DatabaseDataSourceService;
-import com.abixen.platform.service.businessintelligence.multivisualization.service.DomainBuilderService;
+import com.abixen.platform.service.businessintelligence.multivisualization.service.*;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Page;
@@ -28,6 +28,7 @@ import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
 
 import javax.annotation.Resource;
+import java.sql.Connection;
 import java.util.*;
 
 
@@ -43,6 +44,9 @@ public class DatabaseDataSourceServiceImpl implements DatabaseDataSourceService 
 
     @Autowired
     private DomainBuilderService domainBuilderService;
+
+    @Autowired
+    private DatabaseFactory databaseFactory;
 
     @Override
     public Set<DataSourceColumn> getDataSourceColumns(Long dataSourceId) {
@@ -138,5 +142,14 @@ public class DatabaseDataSourceServiceImpl implements DatabaseDataSourceService 
     public DatabaseDataSource findDataSource(Long id) {
         log.debug("findPage() - id: " + id);
         return databaseDataSourceRepository.findOne(id);
+    }
+
+    @Override
+    public List<Map<String, DataValueWeb>> getPreviewData(DatabaseDataSourceForm databaseDataSourceForm) {
+        DatabaseConnection databaseConnection = databaseConnectionService.findDatabaseConnection(databaseDataSourceForm.getDatabaseConnection().getId());
+        DatabaseService databaseService = databaseFactory.getDatabaseService(databaseDataSourceForm.getDatabaseConnection().getDatabaseType());
+        Connection connection = databaseService.getConnection(databaseConnection);
+        List<Map<String, DataValueWeb>> dataSourcePreviewData = databaseService.getDataSourcePreview(connection, buildDataSource(databaseDataSourceForm));
+        return dataSourcePreviewData;
     }
 }

--- a/abixen-platform-business-intelligence-service/src/main/web/admin/businessintelligence/abixen/multi-visualization/modules/database-data-source/html/edit.html
+++ b/abixen-platform-business-intelligence-service/src/main/web/admin/businessintelligence/abixen/multi-visualization/modules/database-data-source/html/edit.html
@@ -77,12 +77,26 @@
         </div>
     </div>
 
+    <div class="row form-row file-preview-table">
+        <div class="col-xs-12">
+            <div
+                    ng-controller="DataBaseDataSourceTableController as dataBaseDataSourceTable">
+                <div ng-if="dataBaseDataSourceTable.renderTable">
+                    <platform-list-grid class="default-platform-list-grid"
+                                        list-grid-config="dataBaseDataSourceTable.listGridConfig"
+                                        load-all-records="true">
+                    </platform-list-grid>
+                </div>
+            </div>
+        </div>
+    </div>
+
 
 </form>
 <div class="row bottom-navigation">
     <div class="col-xs-12">
         <a class="btn save-button"
-           ng-click="databaseDataSourceDetails.testConnection()"><i
+           ng-click="databaseDataSourceDetails.getPreviewData()"><i
                 class="fa fa-table"></i><span>&nbsp;Get Sample Data</span>
         </a>
         <a class="btn save-button"

--- a/abixen-platform-business-intelligence-service/src/main/web/admin/businessintelligence/abixen/multi-visualization/modules/database-data-source/javascript/database-data-source-details.controller.js
+++ b/abixen-platform-business-intelligence-service/src/main/web/admin/businessintelligence/abixen/multi-visualization/modules/database-data-source/javascript/database-data-source-details.controller.js
@@ -27,10 +27,11 @@
         '$stateParams',
         '$log',
         'DatabaseConnection',
+        'DatabaseDataSourcePreviewData',
         'responseHandler'
     ];
 
-    function DatabaseDataSourceDetailsController($scope, DatabaseDataSource, $state, $stateParams, $log, DatabaseConnection, responseHandler) {
+    function DatabaseDataSourceDetailsController($scope, DatabaseDataSource, $state, $stateParams, $log, DatabaseConnection, DatabaseDataSourcePreviewData, responseHandler) {
         $log.log('DatabaseDataSourceDetailsController');
         var databaseDataSourceDetails = this;
 
@@ -52,6 +53,7 @@
         databaseDataSourceDetails.getDatabaseTableColumns = getDatabaseTableColumns;
         databaseDataSourceDetails.beforeSaveForm = beforeSaveForm;
         databaseDataSourceDetails.goToViewMode = goToViewMode;
+        databaseDataSourceDetails.getPreviewData = getPreviewData;
 
         databaseDataSourceDetails.json = null;
         databaseDataSourceDetails.filter = JSON.parse(data);
@@ -160,6 +162,33 @@
             databaseDataSourceDetails.saveForm();
         }
 
+        function getPreviewData() {
+            databaseDataSourceDetails.entity.columns = [];
+
+            var columnPosition = 1;
+
+            for (var i = 0; i < databaseDataSourceDetails.databaseTableColumns.length; i++) {
+                if (databaseDataSourceDetails.databaseTableColumns[i].selected) {
+                    databaseDataSourceDetails.entity.columns.push({
+                        name: databaseDataSourceDetails.databaseTableColumns[i].name,
+                        position: columnPosition++
+                    });
+                }
+            }
+
+            DatabaseDataSourcePreviewData.query({}, databaseDataSourceDetails.entity)
+                .$promise
+                .then(onQueryResult, onQueryError);
+        }
+
+        function onQueryResult(data) {
+            $log.log('DatabaseDataSourcePreviewData.query: ', data);
+            $scope.$broadcast('DatabaseDataSourceDataUpdated', data);
+        }
+
+        function onQueryError(error) {
+            responseHandler.handle(error, $scope);
+        }
 
         function getValidators() {
             var validators = [];

--- a/abixen-platform-business-intelligence-service/src/main/web/admin/businessintelligence/abixen/multi-visualization/modules/database-data-source/javascript/database-data-source-preview.factory.js
+++ b/abixen-platform-business-intelligence-service/src/main/web/admin/businessintelligence/abixen/multi-visualization/modules/database-data-source/javascript/database-data-source-preview.factory.js
@@ -1,0 +1,31 @@
+/**
+ * Copyright (c) 2010-present Abixen Systems. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+(function () {
+
+    'use strict';
+
+    angular
+        .module('platformDatabaseDataSourceModule')
+        .factory('DatabaseDataSourcePreviewData', DatabaseDataSourcePreviewData);
+
+    DatabaseDataSourcePreviewData.$inject = ['$resource'];
+
+    function DatabaseDataSourcePreviewData($resource) {
+        return $resource('/admin/businessintelligence/abixen/multi-visualization/database-data-sources/preview/', {}, {
+            query: {method: 'POST', isArray: true}
+        });
+    }
+
+})();

--- a/abixen-platform-business-intelligence-service/src/main/web/admin/businessintelligence/abixen/multi-visualization/modules/database-data-source/javascript/database-data-source-table.controller.js
+++ b/abixen-platform-business-intelligence-service/src/main/web/admin/businessintelligence/abixen/multi-visualization/modules/database-data-source/javascript/database-data-source-table.controller.js
@@ -1,0 +1,85 @@
+/**
+ * Copyright (c) 2010-present Abixen Systems. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+(function () {
+
+    'use strict';
+
+    angular
+        .module('platformDatabaseDataSourceModule')
+        .controller('DataBaseDataSourceTableController', DataBaseDataSourceTableController);
+
+    DataBaseDataSourceTableController.$inject = [
+        '$scope',
+        '$log'
+    ];
+
+    function DataBaseDataSourceTableController($scope, $log) {
+        $log.log('DataBaseDataSourceTableController');
+
+        var dataBaseDataSourceTable = this;
+        dataBaseDataSourceTable.options = undefined;
+        dataBaseDataSourceTable.data = undefined;
+        dataBaseDataSourceTable.renderTable = false;
+        dataBaseDataSourceTable.fileColumns = [];
+        dataBaseDataSourceTable.gridData = [];
+
+        $scope.$on('DatabaseDataSourceDataUpdated', function (event, data) {
+            $log.debug('DatabaseDataSourceDataUpdated - data: ', data.length);
+            if (data !== undefined && data !== [] && data.length > 0) {
+                dataBaseDataSourceTable.gridData = data;
+                dataBaseDataSourceTable.fileColumns = [];
+                Object.keys(data[0]).forEach(function (column) {
+                    if (column !== undefined && column !== null && column !== '' && column !== '$$hashKey') {
+                        dataBaseDataSourceTable.fileColumns.push({name: column, selected: true});
+                    }
+                });
+                if (dataBaseDataSourceTable.renderTable !== false) {
+                    dataBaseDataSourceTable.listGridConfig.setData(data);
+                }else {
+                    dataBaseDataSourceTable.renderTable = true;
+                }
+            }
+        });
+
+        angular.extend(dataBaseDataSourceTable, new AbstractListGridController(null,
+            {
+                getTableColumns: getTableColumns,
+                dataProviderType: 'list',
+                loadOnStart: false,
+                selectType: 'multi',
+                onTableReady: onTableReady
+
+            }
+        ));
+
+        function getTableColumns() {
+            var columns = [];
+            dataBaseDataSourceTable.fileColumns.forEach(function (column) {
+                columns.push({
+                    field: column.name + ".value",
+                    name: column.name,
+                    cellClass: 'cell-align-right',
+                    enableSorting: false,
+                    visible: column.selected
+                });
+            });
+            return columns;
+        }
+
+        function onTableReady() {
+            dataBaseDataSourceTable.listGridConfig.setData(dataBaseDataSourceTable.gridData);
+        }
+    }
+})();

